### PR TITLE
Add RockBlock SBD satellite modem support

### PR DESCRIPTION
**Pull request by stephendade**
This PR adds support for sending & receiving very basic MAVLink telemetry via a RockBlock SBD satellite modem.

It will allow for literal worldwide telemetry coverage for any vehicle fitted with the aforementioned modem.

The modem bandwidth is very low (~50 bytes per 20sec or so), so I've deliberately limited what can be sent and when (1 HIGH_LATENCY2 packet every 20 sec). I did have to trim the message down (remove MAVLink header and compatibility flag, with are restored at the gateway) a little to keep within 50 bytes.

Requires https://github.com/stephendade/rockblock2mav at the GCS end to send and receive packets.

This PR is WIP, as it requires a bit more testing.

### CODE REVIEW

**Comment by stephendade** [comment]
Happy to assist anyone who wants to test this too

**Comment by auturgy** [comment]
why sending as Mavlink2?  this will never be signed, and mavlink1 has a smaller header

**Comment by stephendade** [comment]
> why sending as Mavlink2? this will never be signed, and mavlink1 has a smaller header

Good point - MAVLink1 is just smaller enough (50 bytes for a ``HIGH_LATENCY2`` message, compared to 52 bytes in MAVLink2) to not require chopping the message up.

I've now updated the PR to use MAVLink1

**Comment by stephendade** [comment]
More updates:

- I've removed the requirement for a second serial port. So only a single scripting port is required.
- Incoming message CRC check added
- Fixed issue where incoming messages would _sometimes_ generate errors

Fairly close to being finished at this port. I intend to start real-life testing with a Cube Orange this week.

**Comment by stephendade** [comment]
This has been successfully tested on a Cube Orange flight controller.

File: libraries/AP_Scripting/generator/description/bindings.desc [hunk 1]
```diff
@@ -555,3 +555,11 @@ userdata uint32_t manual_operator __tostring uint32_t___tostring
...
 userdata uint32_t manual tofloat uint32_t_tofloat
 
+include AP_Scripting/AP_Scripting_MAVLink.h
+singleton ScriptingMAVLink depends HAL_HIGH_LATENCY2_ENABLED
+singleton ScriptingMAVLink rename MAVLink
+singleton ScriptingMAVLink manual begin lua_mavlink_get_MAVLink
+singleton ScriptingMAVLink manual create_high_latency_packet lua_mavlink_create_high_latency_packet
+singleton ScriptingMAVLink manual is_high_latency_enabled lua_mavlink_is_high_latency_enabled
+singleton ScriptingMAVLink manual set_high_latency_enabled lua_mavlink_set_high_latency_enabled
+singleton ScriptingMAVLink manual receive lua_mavlink_receive
```
Comment (IamPete1) [inline libraries/AP_Scripting/generator/description/bindings.desc:565]: I think we could do most of these as auto generated bindings with the exception of `lua_mavlink_get_MAVLink`. The key would be to add `ScriptingMAVLink` as a `ap_object` so we can call it directly.

File: libraries/AP_Scripting/AP_Scripting_MAVLink.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+{
+    //enable or disable high latency mode
+    GCS_MAVLINK *link = gcs().chan(0);
+    mavlink_command_long_t pkt;
+    pkt.param1 = hl_enabled;
+
+    link->handle_control_high_latency(pkt);
+}
+
+uint16_t ScriptingMAVLink::create_high_latency2(uint8_t* msgbuf)
```
Comment (IamPete1) [inline libraries/AP_Scripting/AP_Scripting_MAVLink.cpp:55]: I think this should be done in the script. The ultimate solution would be to directly generate the lua as we do for python and c++.

**Review by IamPete1 — COMMENTED** [review]
We have to be quite careful in how we implement the first direct scripting MAVLink stuff, we don't want to be limited in the future.  

We do also have https://github.com/ArduPilot/ardupilot/pull/13660 that has been sat for a while.

File: libraries/AP_Scripting/AP_Scripting_MAVLink.cpp (hunk 2 above)
Comment (IamPete1) [reply libraries/AP_Scripting/AP_Scripting_MAVLink.cpp:55]: https://github.com/ArduPilot/pymavlink/pull/386

**Comment by stephendade** [comment]
> We do also have https://github.com/ArduPilot/ardupilot/pull/13660 that has been sat for a while.

I did look at that previously. It doesn't do what I need, though. I need to send MAVLink packets through a scripting UART, whereas that PR sends MAVLink packets through a MAVLink UART.

**Comment by IamPete1** [comment]
Sorry, I have not looked at the script in great detail. How does the script interface with ArduPilot? Could it be done in lua stand alone? It looks like there is some command long/int stuff?

**Comment by stephendade** [comment]
Essentially, I want to send and receive a limited set of MAVLink messages over a scripting UART. Only need to send the HIGH_LATENCY2 message and then receive COMMAND_LONG/INT messages.

The fun part is I need to send AT commands before and after each MAVLink message, so the Lua MAVLink functions should accept/return an array of bytes that I send/receive from the scripting port.

**Comment by IamPete1** [comment]
If we were to add a method to forward on command messages into ArduPilot then everything else could be done in lua? Of course the script would be more complex without the C++ helpers, but then we only pay that if your using the script, the extra bindings cost flash for everyone. Looks like this about ~2.3K.

**Comment by stephendade** [comment]
Closing this, as there's a replacement at #22861.

**Pull request closed without merging**